### DISCUSSION TITLE
Blocking mode for Custom Filtering Rules

### DISCUSSION THREAD

**Discussion by sachinwadhwa** [AdguardTeam/AdGuardHome#3431] [Q&A] [answered]
Is it possible to specify blocking mode when specifying custom filtering rules?

I use VPN quite often and internal domain resolution first hits adguard, received NXDOMAIN and then its tried using VPN's DNS servers. I would like to block internal.mycompany.com  but would like to return NXDOMAIN.

When I add custom filtering rules, it returns "default" to internal.mycompany.com, which blocks the page and doesn't retry at VPN's DNS server.

I know I can change default settings to return NXDOMAIN for all blocks but creates 1000's of retry queries by some clients.

**Comment by ameshkov** [comment, marked as answer]
The easiest way would be to use $dnsrewrite modifier:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#dnsrewrite

It'd be something like this:
`||example.org^$dnsrewrite=NXDOMAIN`

**Reply by sachinwadhwa** [reply]
Perfect this works!

Quick one, does this works with regex filters also?

**Reply by ameshkov** [reply]
Sure:
`/example\..*/$dnsrewrite=NXDOMAIN`
